src/stavisky_lab_to_nwb/stavisky/staviskyecephysinterface.py

```diff
@@ -1,23 +1,192 @@
 """Class for converting generic ecephys data."""
-from pynwb import NWBFile
+import json
+import redis
+import numpy as np
+from pynwb import NWBFile, TimeSeries
+from typing import Optional, Union, Literal
+from hdmf.backends.hdf5 import H5DataIO
 
 from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.tools.nwb_helpers import get_module
+from neuroconv.datainterfaces.ecephys.baserecordingextractorinterface import BaseRecordingExtractorInterface
 
 
 class StaviskySpikingBandPowerInterface(BaseDataInterface):
     """Spiking band power interface for Stavisky Redis conversion"""
 
-    def __init__(self):
-        # This should load the data lazily and prepare variables you need
-        pass
+    # TODO: if we decide timestamps need to be aligned with sorting, etc. then
+    # may need to subclass from BaseTemporalAlignmentInterface instead
 
-    def get_metadata(self):
-        # Automatically retrieve as much metadata as possible
-        metadata = super().get_metadata()
+    def __init__(
+        self,
+        port: int,
+        host: str,
+    ):
+        super().__init__(port=port, host=host)
 
-        return metadata
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        chunk_size: int = 1000,
+        timestamp_source: str = "redis",
+        timestamp_encoding: Optional[str] = None,
+        timestamp_dtype: Optional[Union[str, type, np.dtype]] = None,
+        timestamp_unit: Optional[str] = None,
+        smooth_timestamps: bool = False,
+        start_time: Optional[float] = None,
+    ):
+        # Instantiate Redis client and check connection
+        r = redis.Redis(
+            port=self.source_data["port"],
+            host=self.source_data["host"],
+        )
+        r.ping()
 
-    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
-        # All the custom code to write to PyNWB
+        # check timestamp-related args
+        if timestamp_source == "redis":
+            timestamp_unit = "ms"
+            start_time = np.frombuffer(r.xrange("metadata")[0][1][b"startTime"], dtype=np.float64).item()
+        else:
+            assert timestamp_encoding is not None
+            assert timestamp_dtype is not None
+            assert timestamp_unit is not None
+            timestamp_source = bytes(timestamp_source, "utf-8")
+        if timestamp_encoding is not None:
+            assert timestamp_encoding in ["str", "buffer"]
+        if timestamp_unit is not None:
+            assert timestamp_unit in ["s", "ms", "us"]  # TODO: or just directly use scaling factor
```

Comment:
Yeah, I'd just pass the scaling factor. Perhaps a tad less user-friendly, but maximal freedom that way

Reply:
Might be worth indicating, through variable name or docstrings, that all these timestamp things are for the _redis_ timestamps as read from the database, to avoid confusion with the _NWB_ timestamps